tools/rosnode/src/rosnode/__init__.py

```diff
@@ -338,7 +337,7 @@ def rosnode_ping(node_name, max_count=None, verbose=False):
                 # 3786: catch ValueError on unpack as socket.error is not always a tuple
                 try:
                     # #3659
-                    errnum, msg = e
+                    errnum, msg = e.args
```

Comment:
This way allows it to work for both Python 2 & 3. `socket.error` is just an alias of `OSError` in Python 3.